File: DSA/Sarveshwari_Rathore_590025869/Day3/Question2.py
def count(self, nums, target):
        def firstoccurrence():
            left, right = 0, len(nums) - 1
            first = -1
            while left <= right:
                mid = (left + right) // 2
                if nums[mid] == target:
                    first = mid
                    right = mid - 1
                elif nums[mid] < target:
                    left = mid + 1
                else:
                    right = mid - 1
                    return first
        def lastoccurrence():
            left, right = 0, len(nums) - 1
            last = -1
            while left <= right:
                mid = (left + right) // 2
                if nums[mid] == target:
                    last = mid
                    left = mid + 1
                elif nums[mid] < target:
                    left = mid + 1
                else:
                    right = mid - 1
                    return last
        first = firstoccurrence()
        if first == -1:
            return 0
        last = lastoccurrence()
        return last - first + 1
nums = [1,2,3,3,3,4]
target = 3
result = count(nums,target)
print("Occurrences =",result)

File: DSA/Mugdha_590015431/Day_3/Question2.java
package DSA.Mugdha_590015431.Day_3;
import java.util.*;
public class Question2{
    public static int firstOccurrence(int arr[],int target){
        int low=0;
        int high=arr.length-1;
        int first=-1;
        while(low<=high){
            int mid=low+(high-low)/2;
            if(arr[mid]==target){
                first=mid;
                high=mid-1;// search left side
            } else if(arr[mid]<target) {
                low=mid+1;
            } else{
                high=mid-1;
            }
        }
        return first;
    }
    public static int lastOccurrence(int arr[],int target){
        int low=0;
        int high=arr.length-1;
        int last=-1;
        while(low<=high){
            int mid=low+(high-low)/2;
            if(arr[mid]==target){
                last=mid;
                low=mid+1; // search right side
            }
            else if(arr[mid]<target){
                low=mid+1;
            }
            else{
                high=mid-1;
            }
        }
        return last;
    }
    public static int countOccurrences(int arr[],int target){
        int first=firstOccurrence(arr,target);
        if(first==-1){
            return 0;
        }
        int last=lastOccurrence(arr,target);
        return last-first+1;
    }
    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        int arr[]=new int[n];
        for(int i=0;i<n;i++){
            arr[i]=sc.nextInt();
        }
        int target=sc.nextInt();
        System.out.println(countOccurrences(arr,target));
        sc.close();
    }
}
